packages/product/src/loaders/connection.ts

```diff
@@ -62,7 +83,29 @@ async function loadDefault({ database, container }) {
   const entities = Object.values(ProductModels) as unknown as EntitySchema[]
   const orm = await DALUtils.mikroOrmCreateConnection(database, entities)
 
+  return orm.em.fork()
+}
+
+async function loadShared({ container, logger }) {
+  const sharedConnection = container.resolve(
+    ContainerRegistrationKeys.PG_KNEX_CONNECTION_REGISTRATION_KEY,
+    {
+      allowUnregistered: true,
+    }
+  )
+  if (!sharedConnection) {
+    ;(logger ?? console)?.warn(
+      "The Product module is setup to use a shared resources but no shared connection is present. A new connection will be created"
+    )
+  }
+
+  const manager = await loadDefault({
+    database: {
+      connection: sharedConnection,
+      // clientUrl: "",
+    },
+  })
   container.register({
-    manager: asValue(orm.em.fork()),
+    manager: asValue(manager),
```

Comment:
suggestion: All of this is not relevant to just the product module, might be worth to think about moving this to utils eventually.

Reply:
This is planned, i proposed that this morning, you reading my mind 🥹